lib/vector-core/buffers/src/lib.rs

```diff
@@ -27,41 +29,34 @@ use crate::buffer_usage_data::BufferUsageData;
 use crate::bytes::{DecodeBytes, EncodeBytes};
 pub use acker::{Ackable, Acker};
 use core_common::byte_size_of::ByteSizeOf;
-use futures::StreamExt;
-use futures::{channel::mpsc, Sink, SinkExt, Stream};
+use futures::{channel::mpsc, Sink, SinkExt};
+use futures::{Stream, StreamExt};
 use pin_project::pin_project;
 #[cfg(test)]
 use quickcheck::{Arbitrary, Gen};
 use serde::{Deserialize, Serialize};
-use std::fmt::{Debug, Display};
+use std::fmt::Debug;
 use std::pin::Pin;
 use std::sync::Arc;
 use std::task::{Context, Poll};
 use tracing::Span;
 pub use variant::*;
 
+pub type BufferStream<T> = Box<dyn Stream<Item = T> + Unpin + Send>;
+
 /// Build a new buffer based on the passed `Variant`
 ///
 /// # Errors
 ///
 /// This function will fail only when creating a new disk buffer. Because of
 /// legacy reasons the error is not a type but a `String`.
 #[allow(clippy::needless_pass_by_value)]
-pub fn build<'a, T>(
+pub fn build<T>(
     variant: Variant,
     span: Span,
-) -> Result<
-    (
-        BufferInputCloner<T>,
-        Box<dyn Stream<Item = T> + 'a + Unpin + Send>,
-        Acker,
-    ),
-    String,
->
+) -> Result<(BufferInputCloner<T>, BufferStream<T>, Acker), String>
 where
-    T: 'a + ByteSizeOf + Send + Sync + Unpin + Clone + EncodeBytes<T> + DecodeBytes<T>,
-    <T as EncodeBytes<T>>::Error: Debug,
-    <T as DecodeBytes<T>>::Error: Debug + Display,
+    T: Bufferable + Clone + 'static,
```

Comment:
I think `+ 'static` is redundant here, since `Bufferable` already is?

Reply:
You're right.  Fixed in 6a5cce8, along with converting all the other places using the longer type signature to use `Bufferable`.

Reply:
:+1: